module/VuFind/src/VuFind/View/Helper/Root/RecordDataFormatter.php

```diff
@@ -131,20 +151,32 @@ protected function render($driver, $field, $data, $options)
 
     /**
      * Create formatted key/value data based on a record driver and field spec.
+     * The first argument can be a descendant of RecordDriver/AbstractBase.
+     * If omitted, then invoke this class with the desired driver.
+     * The second or first argument is an array containing formatting specifications.
      *
-     * @param RecordDriver $driver Record driver object.
-     * @param array        $spec   Formatting specification
+     * @param array ...$args Record driver object and or formatting specifications.
      *
      * @return array
      */
-    public function getData(RecordDriver $driver, array $spec)
+    public function getData(...$args)
     {
+        if (empty($args[0])) {
+            return [];
+        }
+        if (is_a($args[0], 'VuFind\\RecordDriver\\AbstractBase')) {
```

Comment:
I'd suggest `if ($args[0] instanceof RecordDriver) {` instead -- using the class directly instead of a string is better for static analysis.